test/ref/test.rs

```diff
@@ -99,8 +118,56 @@ async fn install_canister(agent: &Agent, wasm: &[u8], init_arg: &[u8]) -> Princi
     canister_id
 }
 
-#[tokio::main]
-async fn main() {
+fn unzip_file(file_path: &str, output_path: &str) {
+    let file = File::open(file_path).unwrap();
+    let mut decoder = GzDecoder::new(file);
+    let mut output_file = File::create(output_path).unwrap();
+    std::io::copy(&mut decoder, &mut output_file).unwrap();
+}
+
+pub static STATE_MACHINE_BINARY: &str = "../ic-test-state-machine";
+fn sm_env() -> StateMachine {
+    let path = match env::var_os("STATE_MACHINE_BINARY") {
+        None => STATE_MACHINE_BINARY.to_string(),
+        Some(path) => path
+            .clone()
+            .into_string()
+            .unwrap_or_else(|_| panic!("Invalid string path for {path:?}")),
+    };
+
+    if !Path::new(&path).exists() {
+        println!("
+        Could not find state machine binary to run canister integration tests.
+
+        I looked for it at {:?}. You can specify another path with the environment variable STATE_MACHINE_BINARY (note that I run from {:?}).
+
+        Run the following command to get the binary:
+            curl -sLO https://download.dfinity.systems/ic/$commit/binaries/$platform/ic-test-state-machine.gz
+            gzip -d ic-test-state-machine.gz
+            chmod +x ic-test-state-machine
+        where $commit can be read from `.ic-commit` and $platform is 'x86_64-linux' for Linux and 'x86_64-darwin' for Intel/rosetta-enabled Darwin.
+        ", &path, &env::current_dir().map(|x| x.display().to_string()).unwrap_or_else(|_| "an unknown directory".to_string()));
+    }
```

Comment:
These instructions are incorrect for our case; we need to supply the binary via Bazel.